src/item_factory.cpp

```diff
@@ -94,26 +115,32 @@ void Item_factory::finialize_item_blacklist()
     }
 }
 
-void add_to_set(t_string_set &s, JsonObject &json, const std::string &name)
+void add_to_set( t_string_set &s, JsonObject &json, const std::string &name )
 {
-    JsonArray jarr = json.get_array(name);
-    while (jarr.has_more()) {
-        s.insert(jarr.next_string());
+    JsonArray jarr = json.get_array( name );
+    while( jarr.has_more() ) {
+        s.insert( jarr.next_string() );
     }
 }
 
-void Item_factory::load_item_blacklist(JsonObject &json)
+void Item_factory::load_item_blacklist( JsonObject &json )
 {
-    add_to_set(item_blacklist, json, "items");
+    add_to_set( item_blacklist, json, "items" );
 }
 
-void Item_factory::load_item_whitelist(JsonObject &json)
+void Item_factory::load_item_whitelist( JsonObject &json )
 {
-    add_to_set(item_whitelist, json, "items");
+    add_to_set( item_whitelist, json, "items" );
+}
+
+void Item_factory::load_item_option( JsonObject &json )
+{
+    add_to_set( item_options, json, "options" );
 }
 
 Item_factory::~Item_factory()
 {
+    item_options.clear();
```

Comment:
This should probably be in `Item_factory::clear`. The factory is only destroyed when the game is terminated, but `clear` is called before loading the game data.

Btw. why not make `item_options` a member or at least put it into an anonymous namespace/mark it as `static`?